src/ILCompiler.Compiler/src/Compiler/MetadataManager.cs

```diff
@@ -660,7 +660,20 @@ public bool IsReflectionBlocked(TypeDesc type)
                     return IsReflectionBlocked(((ParameterizedType)type).ParameterType);
 
                 case TypeFlags.FunctionPointer:
-                    throw new NotImplementedException();
+                    var signature = ((FunctionPointerType)type).Signature;
+                    if (IsReflectionBlocked(signature.ReturnType))
+                    {
+                        return true;
+                    }
+
+                    for (int i = 0; i < signature.Length; i++)
+                    {
+                        if (IsReflectionBlocked(signature[i]))
+                        {
+                            return true;
+                        }
+                    }
+                    return false;
```

Comment:
Nit: instead of the `if (IsReflectionBlocked(signature.ReturnType))` block above, this could just be `return IsReflectionBlocked(signature.ReturnType)`, saving 5 lines of code.